producto/posintegrado/README.md

```diff
@@ -0,0 +1,158 @@
+Con la Working Key, se cifra el PinBlock y se envía al servidor.
+
+La Working Key se cambia en forma periódica (al menos en cada cierre), para evitar que sea descubierta por
+terceros.
+
+Este modelo de administración de llaves es el que se usará para las llaves MAC.
+
+### El modelo DUKPT – Encriptación de PIN
+
+El nuevo método de administración de llaves para PIN que usará Transbank es el denominado ―Llave Única
+derivada por transacción o DUKPT por sus iniciales en inglés.
+Bajo este método los PED son inicializados en un ambiente seguro, con datos de identificación propios de
+cada PED (Identificador de la llave de derivación, Identificador de PED único y un contador de transacciones
+iniciado en cero), más una llave inicial que se calcula usando los datos propios de cada PED y la llave de
+derivación base. Con esta llave inicial se genera la próxima llave de cifrado para PIN. Este proceso se realiza
+con una función asimétrica (DUKPT del PinPad), es decir, una función de un solo sentido, de forma que el PED
+no sea capaz de generar ninguna llave anterior a la actual.
+
+### Calculo de MAC
+
+Para asegurar la integridad de la información que viaja desde y hacia el Autorizador de comercio, se introduce
+un código de autenticación de mensajes (MAC) el cual es enviado en el mensaje de requerimiento y validado
+por el Autorizador de Transbank al recibirlo. A su vez, el Autorizador de Transbank envía un código de MAC
+para el mensaje de respuesta, el cual debe ser validado por la caja. Si la validación que hace la caja del código
+de MAC es negativa debe generar una reversa. La transacción de reversa debe ser igual a la respuesta recibida
+pero con el campo RESPONSE CODE con el valor 989 y el campo MESSAGE SUBTYPE en R. Cuando el
+Autorizador de Transbank detecta un MAC inválido en el mensaje de requerimiento, envía un mensaje de
+respuesta con código de rechazo 898 (MAC inválido).
+
+### Manejo de llaves MAC (Message Authentication Code)
+
+Las llaves criptográficas para la generación de MAC (working key de MAC) se maneja de acuerdo a lo siguiente:
+Las working key son generadas por el sistema de Transbank y transmitidas en línea para cada uno de los
+terminal ID definidos en el comercio cliente.
+Para la carga y / o cambio de la working keys de MAC se utilizan las transacciones de CIERRE BATCH y CARGA
+DE LLAVE (Ver Transacciones Administrativas).
+
+Las llaves working keys de MAC se actualizan en cada nueva transacción atendida por Transbank. Por lo que
+la caja debe registrar esta nueva llave para su uso en la siguiente transacción.
+Las llaves se deben cambiar automáticamente todos los días. Esto implica que debe existir un procedimiento
+de inicialización o cierre obligatorio en cada caja (terminal ID) que se ejecuta en forma automática todos los
+días y que como parte de este procedimiento se envía a Transbank una transacción de CIERRE BATCH o CARGA
+DE LLAVE por cada caja (terminal ID).
```

Comment:
Wait, so if I do a "Cierre" I don't need a "Carga de Llaves" because the "Cierre" also reloads the keys?

Reply:
Yes, and it also says that with every transaction processed by Transbank it reloads the keys.

Maybe it's a good idea to include the LoadKeys in the initialization after Serial Port is connected.